scripts/check-docs-quality.js

```diff
@@ -16,53 +16,67 @@
  */
 const { execSync, spawnSync } = require('child_process');
 const path = require('path');
+// eslint-disable-next-line import/no-extraneous-dependencies
+const commandExists = require('command-exists');
 
 const listFilesTrackedByGit = 'git ls-files';
 
 const inheritStdIo = {
   stdio: 'inherit',
 };
 
+const LINT_SKIPPED_MESSAGE =
+  'Skipping documentation quality check (vale not found). Install vale linter (https://docs.errata.ai/vale/install) to enable.\n';
 const ERROR_MESSAGE =
-  'Please install vale linter(https://docs.errata.ai/vale/install). Ignore this message if already installed.\n';
+  'Language linter (vale) generated errors. Please check the errors and review any markdown files that you changed.\n';
 
-// xargs is not supported by shx.
-if (process.platform === 'win32') {
-  const validMDFilesCommand = `${listFilesTrackedByGit} | .\\node_modules\\.bin\\shx grep ".md"`;
-  try {
-    // get list of all md files except in directories of gitignore.
-    let filesToLint = execSync(validMDFilesCommand, {
-      stdio: ['ignore', 'pipe', 'inherit'],
-    });
+// Proceed with the script only if Vale linter is installed. Limit the friction and surprises caused by the script.
+commandExists('vale')
+  .catch(() => {
+    console.log(LINT_SKIPPED_MESSAGE);
+    process.exit(0);
+  })
+  .then(() => {
+    // xargs is not supported by shx.
+    if (process.platform === 'win32') {
+      const validMDFilesCommand = `${listFilesTrackedByGit} | .\\node_modules\\.bin\\shx grep ".md"`;
+      try {
+        // get list of all md files except in directories of gitignore.
+        let filesToLint = execSync(validMDFilesCommand, {
```

Comment:
As suggestion, this could leverage the fact that `lint-staged` passes the files to line to the command it invokes. Therefore this subshell could be avoided if there are files in the `argv` array.

For example when `node ./scripts/check-docs-quality` is invoked by lint-staged, it should result in `node ./scripts/check-docs-quality file1.md path/to/file2.md file3.md`

Reply:
Valid suggestion @andrewthauer ! Let me explore this a bit. And TIL `lint-staged` passes the list of files to the command. :)

Reply:
We could add the flag which tracks if this script was called through ```lint-staged``` or manually. If the script is called through ```lint-staged```, we have all valid files in ```argv```. @OrkoHunter Sorry for missing out this case but I can solve it.

Reply:
Hey @Marvin9, what I am thinking is that if the script receives any arguments, it will lint only those files. If the script receives no arguments, it will lint all the (md) files. And no need to apologise at all. :)

Reply:
That's also fine & cleaner solution. In that case[provided files eg. length of argv > 2] we can use spawnSync.

```js
spawnSync('vale', process.argv.slice(2), inheritStdIo);
```